### PR TITLE
fix(secrets): fix file prefix for saml keystore

### PR DESCRIPTION
**Pull request by KathrynLewis**
The saml keystore requires a "file:" prefix in the `gate.yml`, which is currently added in the `secretFileSerializer`.  The bug: when secret decryption is supported in the services AND the keystore file is not an encrypted secret, the code skips the custom secretFileSerializer, which means "file:" doesn't get prepended to the path.  This fixes the logic so that an encrypted secret that should stay encrypted doesn't get the prefix, but all other cases do.

### CODE REVIEW

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/config/v1/secrets/DecryptingObjectMapper.java [hunk 1]
```diff
@@ -107,21 +106,21 @@ public void serialize(Object value, JsonGenerator gen, SerializerProvider provid
...
-                    if (EncryptedSecret.isEncryptedSecret(sValue)) {
+                    if (!EncryptedSecret.isEncryptedSecret(sValue)) {
+                        sValue = annotation.prefix() + sValue;
+                    } else if (shouldDecrypt) {
                         // Decrypt the content of the file and store on the profile under a random
                         // generated file name
                         String name = newRandomFilePath(beanPropertyWriter.getName());
                         byte[] bytes = secretSessionManager.decryptAsBytes(sValue);
                         profile.getDecryptedFiles().put(name, bytes);
                         sValue = getCompleteFilePath(name);
```
Comment (dibyom) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/config/v1/secrets/DecryptingObjectMapper.java]: Can this simply be `sValue = annotation.prefix() + getCompleteFilePath(name);` ?

Comment (KathrynLewis) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/config/v1/secrets/DecryptingObjectMapper.java]: yes, will change!

**Comment by dibyom** [comment]
Fixes https://github.com/spinnaker/spinnaker/issues/4294

**Comment by dibyom** [comment]
@spinnakerbot  cherry-pick 1.19

**Comment by dibyom** [comment]
@spinnakerbot cherry-pick 1.19

**Pull request merged**